### PR TITLE
Email validation for adding team members and contacts

### PR DESCRIPTION
**Pull request by ashleysmithTTD**
What Changed:
- Better validation for email when adding Team Member, adding Business Contact, adding new Participant
- Error message in dialog if user tries to enter a team member email that already exists for the given participant
- Updated error messages based on Gen's feedback

Test Plan:
- Enter valid and invalid emails for Add Team Member dialog, Add Business Contact dialog, Edit Business Contact Dialog, Add new participant in Contact Email field
- Add a team member that already exists
- Add a team member that exists in your portal db but not in the given participant and make sure that still gives a 400 error

### CODE REVIEW

File: src/web/components/BusinessContacts/BusinessContactDialog.tsx [hunk 1]
```diff
@@ -47,7 +47,7 @@ function BusinessContactDialog({
             rules={{
               required: 'Please specify email alias.',
               pattern: {
-                value: /\S+@\S+\.\S+/,
+                value: /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/,
```
Comment (alex-yau-ttd) [inline src/web/components/BusinessContacts/BusinessContactDialog.tsx:50]: Can we extract this to a variable? I'm sure I'm not alone in struggling to understand regex :D

File: src/web/components/TeamMember/TeamMemberHelper.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { UserDTO } from "../../../api/entities/User";
+
+export const validateUniqueTeamMemberEmail = (
+  value: string,
+  existingTeamMembers: UserDTO[] | undefined
```
Comment (alex-yau-ttd) [inline src/web/components/TeamMember/TeamMemberHelper.ts:5]: Can't we remove the `undefined` here?
```
existingTeamMembers: UserDTO[]
```

Comment (alex-yau-ttd) [reply src/web/components/TeamMember/TeamMemberHelper.ts:5]: Then line 7 becomes:
```
if (existingTeamMembers.filter((t) => t.email === value).length > 0) {
```

**Review by alex-yau-ttd — APPROVED** [review]

File: src/web/components/BusinessContacts/BusinessContactDialog.tsx (hunk 1 above)
Comment (ashleysmithTTD) [reply src/web/components/BusinessContacts/BusinessContactDialog.tsx:50]: Variable extracted and reused

**Pull request merged**